sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, find_cycle, is_directed_acyclic_graph
+from sql_directory_parser import SqlFile
+
+
+class SqlFilesDAG:
+    """A DAG of sql files i.e. used for finding the right order to execute the sql files in."""
+
+    @staticmethod
+    def build(sql_files: set[SqlFile]) -> list[SqlFile]:
+        """
+        Build a directed graph from a list of sql files.
+
+        :params sql_files: The sql files to use for DAG generation.
+
+        :returns: a list of sql files sorted by least dependencies first
+            so they can be called sequentially in python code.
+        """
+        graph = DiGraph()
+
+        for sql_file in sql_files:
+            graph.add_node(sql_file.get_variable_name())
+            for parameter in sql_file.get_parameters():
+                graph.add_edge(sql_file.get_variable_name(), parameter)
```

Comment:
This line seems... wrong? Like we need to make sure that the variable referenced is actually a table before we add it to the graph.

Reply:
Not sure what you mean. Every `parameter` is an input table and `sql_file.get_variable_name()` [equals](https://github.com/astronomer/astro-sdk/blob/cb8f6de7c6c64130f8c04a11798bcf6d23a5e1d8/sql-cli/src/sql_cli/template.jinja2#L17) the output table name in the **current** implementation.